Session1/EFCoreAssignment/Program.cs

```diff
@@ -82,3 +81,82 @@ static void LoadingRelatedData_EagerLoading(AppDbContext dbContext)
 }
 
 Console.WriteLine("EF Core is the best");
+
+// TODO: Assignment 2
+
+InsertProduct(new AppDbContext(options));
+InsertProductWithNewShop(new AppDbContext(options));
+UpdateProduct(new AppDbContext(options));
+DeleteProduct(new AppDbContext(options));
+DeleteProductByKey(new AppDbContext(options));
+
+static void InsertProduct(AppDbContext dbContext)
+{
+    // TODO: Insert a new Product
+    using (dbContext)
+    {
+        var product = new Product()
+        {
+            Name = "Product 6",
+            ShopId = 1
+        };
+
+        dbContext.Add(product);
+        dbContext.SaveChanges();
+    }
+}
+
+static void InsertProductWithNewShop(AppDbContext dbContext)
+{
+    // TODO: Insert a new Product with a new Shop
+    var shop = new Shop()
+    {
+        Name = "Shop 4"
+    };
+
+    var product = new Product()
+    {
+        Name = "Product 7",
+        Shop = shop
+    };
+
+    dbContext.Add(shop);
```

Comment:
No need to include this code **dbContext.Add(shop);**. Shop is already being track by EF since it is a navigational property.

Reply:
Done. Thank you for clarifying this!